src/lib/modes/aead/aead.h

```diff
@@ -116,8 +122,7 @@ class BOTAN_PUBLIC_API(2,0) AEAD_Mode : public Cipher_Mode
       *
       * @param ad the associated data
       */
-      template<typename Alloc>
-      void set_ad(const std::vector<uint8_t, Alloc>& ad)
+      void set_ad(std::span<const uint8_t> ad)
```

Comment:
Should that be deprecated as well, on the same basis as `set_associated_data_vec`. Its just an alias for `set_associated_data()`.

Reply:
Kind of an alias, but kind of not. The point of this is to bypass exactly the virtual overloading problem you encountered. Otherwise they all would have just been named `set_associated_data`.

Reply:
With the (protected) `set_ad_n` described in the other comment, they can all be called `set_associated_data` now. Hence the suggestion to deprecate it.

Reply:
Yeah fine to deprecate given the other changes in this PR